tests/test/scheduler/test_exec_graph.cpp

```diff
@@ -72,6 +72,25 @@ TEST_CASE("Test execution graph", "[scheduler][exec-graph]")
     checkExecGraphEquality(expected, actual);
 }
 
+TEST_CASE("Test execution graph node count", "[scheduler][exec-graph]")
```

Comment:
This is very nitpicky but the API is about getting the unique host names from the graph, not the count. 

I'd have thought this test would be called `Test get unique hosts from exec graph`, then the assertion at the end would look like:

```cpp
std::set<std::string> expected = {"foo", "bar", "baz"};
REQUIRE(hosts == expected);
```

Currently the test also doesn't cover duplicates, it's got 3 messages each executed on a different host. It would be good to add another message for a different function but executed on the same host as one of the others to be extra sure.

Reply:
All good points, thanks.